discovery_node/substrate-ingest/src/index.ts

```diff
@@ -0,0 +1,54 @@
+/* eslint-disable @typescript-eslint/require-await */
+/* eslint-disable @typescript-eslint/unbound-method */
+
+// Required imports
+
+import { Command } from 'commander';
+
+import Config from './Config';
+import { StateKeeper, State } from './state/StateKeeper'
+import JoyNode from './joynode/QueryNode'
+import ESUploader from './esearch/ESUploader'
+
+const logger = require('log4js').getLogger('main')
+const command = new Command();
+
+async function main() {
+    command
+        .version('0.0.1')
+        .description("Joystream ElasticSearch import tool")
+        .option('-c, --config <file>', 'path to config', 'config.yml')
+        .option('-d --debug', 'debug')
+        .parse(process.argv);
+
+    const config = new Config(command.config);
+    // TODO: move to a sep init class, catch init errors
```

Comment:
I agree with this TODO, the setup code here is non-trivial, and not easily testable.